src/EnergyPlus/UnitarySystem.cc

```diff
@@ -0,0 +1,13534 @@
+                }
+                thisDesignSpec.noLoadAirFlowRateRatio = loc_noLoadAirFlowRateRatio;
+
+                thisDesignSpec.heatingVolFlowRatio.resize(maxSpeeds);
+                thisDesignSpec.coolingVolFlowRatio.resize(maxSpeeds);
+
+                auto speedFlowRatios = fields.find("flow_ratios");
+                if (speedFlowRatios != fields.end()) {
+                    auto flowRatioArray = speedFlowRatios.value();
+                    int numSpeedInputs = flowRatioArray.size();
+                    if (numSpeedInputs == maxSpeeds) {
+                        int speedNum = -1;
+                        for (auto flowRatio : flowRatioArray) {
+                            speedNum += 1;
+                            auto coolingSpeedRatioObject = flowRatio.at("cooling_speed_supply_air_flow_ratio");
+                            if (coolingSpeedRatioObject == "Autosize") {
+                                thisDesignSpec.coolingVolFlowRatio[speedNum] = -99999;
+                            } else {
+                                thisDesignSpec.coolingVolFlowRatio[speedNum] = coolingSpeedRatioObject;
+                            }
+                            auto heatingSpeedRatioObject = flowRatio.at("heating_speed_supply_air_flow_ratio");
+                            if (heatingSpeedRatioObject == "Autosize") {
+                                thisDesignSpec.heatingVolFlowRatio[speedNum] = -99999;
+                            } else {
+                                thisDesignSpec.heatingVolFlowRatio[speedNum] = heatingSpeedRatioObject;
+                            }
+                        }
+                    } else if (numSpeedInputs < maxSpeeds) {
+                        ShowSevereError(cCurrentModuleObject + ": Error getting inputs for system named: " + thisObjectName);
+                        ShowContinueError("Number of speed inputs (" + General::TrimSigDigits(Real64(numSpeedInputs), 0) +
+                                          " is less than number of speeds (" + General::TrimSigDigits(Real64(maxSpeeds), 0) + ").");
+                        errorsFound = true;
+                    } else {
+                        ShowSevereError(cCurrentModuleObject + ": Error getting inputs for system named: " + thisObjectName);
+                        ShowContinueError("Number of speed inputs (" + General::TrimSigDigits(Real64(numSpeedInputs), 0) +
+                                          " is greater than number of speeds (" + General::TrimSigDigits(Real64(maxSpeeds), 0) + ").");
+                        errorsFound = true;
+                    }
+                }
+                designSpecMSHP.push_back(thisDesignSpec);
+            }
+        }
+    } // namespace UnitarySystems
+
+    UnitarySys *UnitarySys::factory(int object_type_of_num, std::string const objectName)
+    {
+        if (UnitarySystems::getInputOnceFlag) {
```

Comment:
@Myoldmopar if you do try to run this you will have to comment out line 479 (getInputOnceFlag) since getInput now only gets 1 unitary system. Maybe that has something to do with my pointer issue?